common/djangoapps/student/views/login.py

```diff
@@ -386,6 +386,28 @@ def send_reactivation_email_for_user(user):
     return JsonResponse({"success": True})
 
 
+@ensure_csrf_cookie
+def verify_user_password(request):
```

Comment:
At the moment, anonymous requests don't generate an exception. They just get passed to authenticate with the anonymous user and get returned a 403 because the password isn't valid. I am happy to change the behavior if it's better to be doing something else.

Reply:
I also favor adding a `login_required` decorator to this endpoint - for extra security.  We expect this to be called only by logged in users anyway.